### ISSUE TITLE
Stopwords not working properly for Kurdish

### ISSUE THREAD

**Issue by Xoshbin** [Donatello-za/rake-php-plus#18]
Hi, After merging my pull request I found out that the list not working properly as expected, For example this character '،' which it's the first line in the Kurdish list is not excluded when Im trying to extract a list of words from a paragraph, For example now that the Corona virus is a trend, in Kurdish for example this word "کۆرۆنا،" which means "Corona," the "،" character is supposed to be detached from the word, but it's not.
I hope I'm clear what I mean!

**Comment by Donatello-za** [comment]
Hi @Xoshbin, I think I understand what you are saying. Does the "،" character have a space between itself and the word "کۆرۆنا،" in the original text? I think the library will also have trouble even in English when a stop word is "attached" to another word. For example, take the stop word "and", which may be found in a person's name, for example "André" or android, it can't remove that from the string as you'll end up with only "ré" or "roid". The library relies on spaces and other symbols separating the words from one another.

**Comment by Xoshbin** [comment]
Hi @Donatello-za thanks for replying, No there's no space between the character and the word, That's why I wrote it like above to let you know that there's no space between them. Anyway since you say that it's not possible to make it work that way because of the words that contain some words in the list, Is it possible to separate special characters like (? , ! . ; :) "and the rest of the other characters" to be detached form the words? Thanks

**Comment by Donatello-za** [comment]
The text is always split into sentences first, take a look at it [here](https://github.com/Donatello-za/rake-php-plus/blob/99460a26404c6ab2a4dc4d8aa3cd402f1865e16a/src/RakePlus.php#L307). 

As you can see, the code assumes typically used punctuation marks that are used in English and is unfortunately hard coded. Perhaps one should be able to override the regular expression to define your own set of special characters with which to split sentences. But we'll have to take a good look at it as this will be language dependent and so I don't want to make a rush decision on how to approach this problem. Perhaps you can help me understand a bit more about the Kurdish langage, for example, is the "،" character used as punctuation of some sort?

**Comment by Xoshbin** [comment]
Whereas There's not much different between the special character in Kurdish and English as you say the "،" character is used for the same purpose as "," in english. But I will try to make a dictionary of Kurdish special characters and it's equivalent in English when I got time.

**Comment by Donatello-za** [comment]
Thanks, that will help @Xoshbin. Just a note, if the "،" character is the same as a comma in English then it also shouldn't be part of the stopwords file so I will remove it when you've send me the list of special characters.

**Comment by Xoshbin** [comment]
@Donatello-za That's the "dictionary" of Kurdish Special characters and it's equivalent in English. As far as I know that's all, If there's any new characters I will keep it updated by myself after I know where you put these new characters. And if you need any help as removing the special characters in the Kurdish List please let me know. 
```
Kurdish            English
-------            -------
،                      ,
؟                      ?
؛                      ;
```

For other special characters like ( . and - “ ) characters, In view it looks like the same but based on this [page](https://www.fileformat.info/info/unicode/char/search.htm?q=-&preview=entity) it looks like in unicode it's different for most of the languages. Whereas there's no Kurdish unicode for some of these characters but the Arabic characters may work for Kurdish too.
I'm saying these special characters because I already see in my application that they are been bypassed by the library and not detached form the words. So we may have to update the special characters by 
the time when we see the library in action.

**Comment by Donatello-za** [comment]
Hi @Xoshbin, I implemented a change that should fix the problems you are having in both Kurdish and Arabic. Please read my whole post below with regards to the issue. If you want to test it before I officially release it you can change your composer.json to point directly to the development beta:

```json
"require": {
        "donatello-za/rake-php-plus": "dev-v1.0.13-dev"
    }
```
You don't have to change anything with regards to how you run your code, it should work out of the box. However, if you are interested in the unit tests I created for Kurdish please see [this file](https://github.com/Donatello-za/rake-php-plus/blob/v1.0.13-dev/tests/lang_ckb_IQ_Test.php).

Also take a look at which characters will be [stripped here](https://github.com/Donatello-za/rake-php-plus/blob/e311da4a4d8196fcead915143620462b289c2d45/src/LangParseOptions.php#L36).

**Important:** One thing I was not sure about is whether to include the "-" (dash / hyphen) character as well because in many languages it is used to join words together and so I can't include that for English, etc. However, I did add it for both Kurdish and Arabic as I noticed it doesn't seem to join words together, am I correct in this? This means it will also remove that character in your case.

Let me know what you think.

**Comment by Xoshbin** [comment]
Thanks @Donatello-za , I just put the dev release in production test. It may take some days to see the results. And for the tests it's great thanks. For the "-" character even I'm not an expert in languages but I see it it's used for joining words in English but I never saw it used for the same purpose in Kurdish and not sure about Arabic too sorry. But still it's place should be where it's right [now](https://github.com/Donatello-za/rake-php-plus/blob/e311da4a4d8196fcead915143620462b289c2d45/src/LangParseOptions.php#L36) . Thank you

**Comment by Donatello-za** [comment]
Cool, I think it is safe to release it and I will do it first thing
tomorrow morning.

**Comment by Donatello-za** [comment]
Hi @Xoshbin, I've released v1.0.13 and will close the issue for now. If there are any specific issues directly related to this I will re-open the issue again.
